frontend/sw.js

```diff
@@ -27,6 +29,17 @@
         await myCache.addAll(urls);
     }
 
+    /**
+     * Clear old version caches.
+     */
+    function clearOldCache() {
+        caches.keys().then((keys) => {keys.filter((key) => !key.includes(CACHE_SUFIX)).map((key) => caches.delete(key))});
+    }
+
+    self.addEventListener('activate', () => {
+        clearOldCache();
```

Comment:
Receive event as a parameter and add event.waitUntil(clearOldCache()) so that the activation ends only when the cache is cleared.